### PR TITLE
Migrate add-on to v2 format & upgrade to Ember.js v5.4.1

### LINKED ISSUE #279: README.md is not visible on NPM
With the migration to V2 addon, the `README.md` is no longer visible on NPM because we publish only the addon `dist`. The README.md should be visible on both GitHub and NPM page.

### PR DESCRIPTION
**Pull request by MrChocolatine**
### Migrate add-on to v2 format (#299)

The add-on is now a full v2 add-on, it was recreated as such using Embroider and the latest standard and specifications.

```
ember addon myaddon --blueprint @embroider/addon-blueprint --typescript --yarn
```

- Blueprint for scaffolding ember v2 addons
  https://github.com/embroider-build/addon-blueprint

### Upgrade add-on to Ember.js v5.4.1 (#299)

- Ember.js v4.8 or above is now required
- Embroider or `ember-auto-import` v2 are now required

---

Closes #279 .

---

_Below is just extra text to give more context, but it should not be included in our changelog._

## Actions taken

1. Fresh add-on created using the latest version of Ember.js + flag `--typescript` + v2 blueprint from Embroider
1. Then re-import everything from our add-on
1. Re-adjust a few things to make it work for our setup

## Resources

- Guide: Porting an add-on to V2
  https://github.com/embroider-build/embroider/blob/main/docs/porting-addons-to-v2.md

### CODE REVIEW

File: test-app/package.json
Comment (MrChocolatine) [inline test-app/package.json:1]: Test application, which is internal and what the end users will use, so we don't need any of the usual information like `author`, `description` etc etc...

**Comment by MrChocolatine** [comment]
@DazzlingFugu/emberenos

**Comment by MrChocolatine** [comment]
@DazzlingFugu/emberenos

**Comment by MrChocolatine** [comment]
@DazzlingFugu/emberenos

**Comment by MrChocolatine** [comment]
We have to move forward @DazzlingFugu/emberenos , please review.
It has been 3 weeks.

**Comment by MrChocolatine** [comment]
We have to move forward @DazzlingFugu/emberenos , please review.
It has been 3 weeks.

**Review by Pixelik — APPROVED** [review]
Couldn't manage to install locally and test but as long as automated tests are passing I'm fine with unblocking this 👍

**Pull request merged**